checks.c

```diff
@@ -1348,6 +1348,11 @@ static void CheckSerial(X509 *x509)
 static void CheckPublicKey(X509 *x509, struct tm tm_after)
 {
 	EVP_PKEY *pkey = X509_get_pubkey(x509);
+	if (pkey == NULL)
+	{
+		SetError(ERR_INVALID);
```

Comment:
It really should be an error, but then for being an unknown type.  And I guess it should also give that error if the type != RSA and EC